aptos-move/framework/aptos-stdlib/sources/hash.move

```diff
@@ -23,6 +23,14 @@ module aptos_std::aptos_hash {
 
     native public fun keccak256(bytes: vector<u8>): vector<u8>;
 
+    native public fun sha2_512(bytes: vector<u8>): vector<u8>;
+
+    native public fun sha3_512(bytes: vector<u8>): vector<u8>;
+
+    /// WARNING: Only 80-bit security is provided by this function. This means an adversary who can compute roughly 2^80
+    /// hashes will, with high probability, find a collision x_1 != x_2 such that RIPEMD-160(x_1) = RIPEMD-160(x_2).
+    native public fun ripemd160(bytes: vector<u8>): vector<u8>;
+
```

Comment:
per earlier conversation, do we want to instead make these internal and provide Move functions that call into these... I don't know where we left off

also what about feature flags?

Reply:
Yes, we will have to because there's no other way to feature-gate them, right @vgao1996 / @wrwg?